Tasks/Common/nuget-task-common/NuGetToolRunner2.ts

```diff
@@ -41,31 +45,59 @@ function prepareNuGetExeEnvironment(
             }
         }
 
+        // New credential provider
+        if (e.toUpperCase() === "NUGET_PLUGIN_PATHS") {
+            pluginCredentialProviderPath = input[e];
+            continue;
+        }
+
+        // Old credential provider
         if (e.toUpperCase() === "NUGET_CREDENTIALPROVIDERS_PATH") {
             originalCredProviderPath = input[e];
+            continue;
+        }
 
-            // will re-set this variable below
+        if (e.toUpperCase() === "VSS_NUGET_URI_PREFIXES") {
+            prefix = input[e];
             continue;
         }
 
         env[e] = input[e];
     }
 
-    let credProviderPath = settings.credProviderFolder || originalCredProviderPath;
-    if (settings.credProviderFolder && originalCredProviderPath) {
-        credProviderPath = settings.credProviderFolder + ";" + originalCredProviderPath;
-    }
-
-    if (authInfo && authInfo.internalAuthInfo){
+    let prefixes: string = null;
+    if (authInfo && authInfo.internalAuthInfo) {
         env["VSS_NUGET_ACCESSTOKEN"] = authInfo.internalAuthInfo.accessToken;
-        env["VSS_NUGET_URI_PREFIXES"] = authInfo.internalAuthInfo.uriPrefixes.join(";");
+        prefixes = authInfo.internalAuthInfo.uriPrefixes.join(";");
+        env["VSS_NUGET_URI_PREFIXES"] = prefixes;
     }
 
     env["NUGET_CREDENTIAL_PROVIDER_OVERRIDE_DEFAULT"] = "true";
 
-    if (credProviderPath) {
-        tl.debug(`credProviderPath = ${credProviderPath}`);
-        env["NUGET_CREDENTIALPROVIDERS_PATH"] = credProviderPath;
+    // If NUGET_CREDENTIALPROVIDERS_PATH is provided and NUGET_PLUGIN_PATHS is not, 
+    // use old cred provider regardles of which version on NuGet
+    if ((settings.quirks && settings.quirks.hasQuirk(NuGetQuirkName.V2CredentialProvider) === false)
+        || (originalCredProviderPath != null && pluginCredentialProviderPath === null)
```

Comment:
All this logic shouldn't be here and be moved to `isCredentialProviderEnabled` (which disables the V1 provider if NuGet >= 4.8.0) and perhaps create a `isCredentialProviderV2Enabled` that enables it for NuGet >= 4.8.0. This will also allow us to use an env var to override which plugin to use in case we need to.

Reply:
Moved